lib/galaxy/jobs/runners/__init__.py

```diff
@@ -313,13 +313,36 @@ def write_executable_script( self, path, contents, mode=0o755 ):
         with open( path, 'w' ) as f:
             f.write( contents )
         os.chmod( path, mode )
-        try:
-            # sync file system to avoid "Text file busy" problems.
-            # These have occurred both in Docker containers and on EC2 clusters
-            # under high load.
-            subprocess.check_call(["sync"])
-        except Exception:
-            pass
+        self._handle_script_integrity( path )
+
+    def _handle_script_integrity( self, path ):
+        if not getattr( self.app.config, "check_job_script_integrity", True ):
```

Comment:
I think that would be a mistake - that is a premature optimization and violation of a good abstraction. ``write_executable_script`` should not need to know anything at all about script integrity is checked or under what conditions it is checked - the ``_handle_script_integrity`` abstraction should be responsible that.